### DISCUSSION TITLE
Cant understand the problem here!!

### DISCUSSION THREAD

**Discussion by TechnicallyKiller** [Cyfrin/foundry-full-course-cu#2949] [Q&A] [answered]
//FUNCTION CALL
 function testEmitsEventOnEntrance() public {
    // Arrange
    vm.prank(PLAYER); // Set PLAYER as the sender of the transaction

    // Act / Assert
    // Expect the 'EnteredRaffle' event to be emitted with the PLAYER address as the indexed parameter
    vm.expectEmit(true, true, false, false, address(raffle)); // Expect the event from the raffle contract
    emit EnteredRaffle(PLAYER); // Emitting the event manually for comparison
    raffle.enterRaffle{value: entranceFee}(); // Call the function to enter the raffle
}

//OUTPUT
Ran 1 test for test/unit/RaffleTest1.sol:RafflesTest
[FAIL: log != expected log] testEmitsEventOnEntrance() (gas: 68584)
Traces:
  [68584] RafflesTest::testEmitsEventOnEntrance()
    ├─ [0] VM::prank(player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    │   └─ ← [Return] 
    ├─ [0] VM::expectEmit(true, true, false, false, Raffle: [0xA8452Ec99ce0C64f20701dB7dD3abDb607c00496])
    │   └─ ← [Return] 
    ├─ emit EnteredRaffle(player: player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    ├─ [47765] Raffle::enterRaffle{value: 10000000000000000}()
    │   ├─ emit IndexedMembers(player: player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    │   └─ ← [Stop] 
    └─ ← [Revert] log != expected log

Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 9.15ms (681.80µs CPU time)

Ran 1 test suite in 15.07s (9.15ms CPU time): 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/unit/RaffleTest1.sol:RafflesTest
[FAIL: log != expected log] testEmitsEventOnEntrance() (gas: 68584)

Encountered a total of 1 failing tests, 0 tests succeeded

**Comment by EngrPips** [comment]
The below is the stack trace of my test that passed, and the difference between my stack trace is that my `Lottery.sol` contract emitted the same event that I expected, but yours is emitting a different event from what you expected.

```js
Traces:
  [68629] LotteryTest::testLotteryProperlyEmitEventWhenAPlayerEnters()
    ├─ [0] VM::expectEmit(true, false, false, false, Lottery: [0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0])
    │   └─ ← [Return] 
    ├─ emit enteredLottery(player: player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    ├─ [0] VM::prank(player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    │   └─ ← [Return] 
    ├─ [47805] Lottery::enter{value: 10000000000000000}()
    │   ├─ emit enteredLottery(player: player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    │   └─ ← [Stop] 
    └─ ← [Stop] 
```

**Comment by NikolayPIvanov** [comment, marked as answer]
Events in Solidity have signatures. The signature of an event is derived from its name and the types of its parameters. This signature allows events to be uniquely identified in the Ethereum Virtual Machine (EVM).

Looking at your code, the event used in Raffle.sol is `IndexedMembers`, whereas in the tests it is `EnteredRaffle` which is a different signatures just from the name.

Placing the same event name `IndexedMembers` makes the test pass.

```solidity
Traces:
  [73092] RafflesTest::testEmitsEventOnEntrance()
    ├─ [0] VM::prank(player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    │   └─ ← [Return] 
    ├─ [0] VM::recordLogs()
    │   └─ ← [Return] 
    ├─ [0] console::log("PLAYER Address: ", player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C]) [staticcall]
    │   └─ ← [Stop] 
    ├─ [0] console::log("Entrance Fee: ", 10000000000000000 [1e16]) [staticcall]
    │   └─ ← [Stop] 
    ├─ [0] VM::expectEmit(true, false, false, false, Raffle: [0x90193C961A926261B756D1E5bb255e67ff9498A1])
    │   └─ ← [Return] 
    ├─ emit IndexedMembers(player: player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    ├─ [47765] Raffle::enterRaffle{value: 10000000000000000}()
    │   ├─ emit IndexedMembers(player: player: [0x44E97aF4418b7a17AABD8090bEA0A471a366305C])
    │   └─ ← [Stop] 
    └─ ← [Stop]
```

**Reply by EngrPips** [reply]
Great heads-up.

**Reply by TechnicallyKiller** [reply]
Thank you Sir!!
